lib/tasks/bulk.rake

```diff
@@ -325,4 +327,45 @@ namespace :bulk do
     attrs = { append_collection_ids: Valkyrie::ID.new(append_coll), skip_validation: true }
     BulkEditService.perform(collection_id: Valkyrie::ID.new(coll), attributes: attrs, metadata_adapter: Valkyrie::MetadataAdapter.find(:indexing_persister), logger: logger)
   end
+
+  desc "Ingest resources from a csv file"
+  task from_csv: :environment do
+    basedir = ENV["BASEDIR"]
+    coll = ENV["COLL"]
+    csvfile = ENV["CSV"]
+
+    abort "usage: COLL=collection_id BASEDIR=directory CSV=csvfile rake bulk:from_csv" unless coll && basedir
+    abort "no such file #{csvfile}" unless File.file?(csvfile)
+
+    user = User.find_by_user_key(ENV["USER"]) if ENV["USER"]
+    user = User.all.select(&:admin?).first unless user
+    class_name = "ScannedResource"
+    @logger = Logger.new(STDOUT)
+
+    begin
+      csv = CSV.read(csvfile, headers: true)
+    rescue => e
+      @logger.error "Error: #{e.message}"
+      @logger.error e.backtrace
+    end
+
+    logger.info "processing #{csv.length} rows"
+    csv.each do |row|
+      logger.info "processing #{row}"
+      attrs = {}
```

Comment:
This could be simplified to:
```
attrs = row.map { |k, v| [k.to_sym, v] }.to_h
```

Reply:
As an FP afficionado I should have seen that!  Done.